packages/core/src/animation/internal/AnimatorStatePlayData.ts

```diff
@@ -28,12 +28,25 @@ export class AnimatorStatePlayData {
     let time = this.frameTime;
     const duration = state._getDuration();
     this.playState = AnimatorStatePlayState.Playing;
-    if (time > duration) {
-      if (state.wrapMode === WrapMode.Loop) {
-        time = time % duration;
-      } else {
-        time = duration;
-        this.playState = AnimatorStatePlayState.Finished;
+    if (time > 0) {
+      if (time > duration) {
+        if (state.wrapMode === WrapMode.Loop) {
+          time = time % duration;
+        } else {
+          time = duration;
+          this.playState = AnimatorStatePlayState.Finished;
+        }
+      }
+    }
+    if (time < 0) {
+      time = duration+ time;
+      if (time < 0) {
+        if (state.wrapMode === WrapMode.Loop) {
+          time = duration + (time % duration);
+        } else {
+          time = 0;
+          this.playState = AnimatorStatePlayState.Finished;
+        }
       }
```

Comment:
Simplified logic like this:
**Important: When adding features, you should re-integrate and simplify the logic**
```ts
if (state.wrapMode === WrapMode.Loop) {
      time = time % duration;
      time < 0 && (time += duration);
    } else {
      if (Math.abs(time) > duration) {
        time = duration;
        this.playState = AnimatorStatePlayState.Finished;
      }
    }
```

Reply:
> Simplified logic like this: **Important: When adding features, you should re-integrate and simplify the logic**
> 
> ```ts
> if (Math.abs(time) > duration) {
>  time = duration;
>}
> ```
When wrapMode !== WrapMode.Loop, i think time should be 0